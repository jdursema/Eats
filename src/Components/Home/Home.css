.home {
  text-align: center;
  background-image: url("../../assets/Food.jpg");
  background-position: right;
  background-repeat: no-repeat;
  height: 100vh;
  margin-top: -100px;
  margin-right: 20px;
  display: flex;
  justify-content: center;
  align-items: left;
  flex-direction: column;
  background-color: #0F2130; }
  .home h1 {
    font-family: 'IM Fell English SC', serif;
    font-size: 100px; }
  .home h3 {
    color: white;
    font-family: 'Raleway', sans-serif;
    width: 30%; }
