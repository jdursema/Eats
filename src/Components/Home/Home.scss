.home{
  text-align: center;
  background-image: url('../../assets/Food.jpg');
	background-position: right;
  background-repeat: no-repeat;
  height: 100vh;
  margin-top: -100px;
  margin-right: 20px;
  display: flex;
  justify-content: center;
  align-items: left;
  flex-direction: column;
  background-color:  #0F2130;

  h1{
    font-family: 'IM Fell English SC', serif;
    font-size: 100px;
  }
  h3{
    color: white;
    font-family: 'Raleway', sans-serif;
    width: 30%;

  }
}
