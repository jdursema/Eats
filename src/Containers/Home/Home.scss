.home{
  text-align: center;
  background-image: url('../../assets/Food.jpg');
	background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 80vh;
  display: flex;
  justify-content: center;
  align-items: left;
  flex-direction: column;
  background-color:  #0F2130;

  .home-text{
    background-color: #0F2130;
    opacity: .8;
    display: flex;
    flex-direction: column;
    align-items: left;
    width: 40%;
    margin: 1%;
    padding: 1%;
  }

  h1{
    font-family: 'IM Fell English SC', serif;
    font-size: 75px;
    color: white;
    opacity: 10;
    padding: 0;
    margin: 0;
    line-height: 1.3;
  }
  h3{
    color: white;
    font-family: 'Raleway', sans-serif;
  }

  .login{
    border: white 2px solid;
    padding: 1%;
    width: 100px;
    color: white;
    text-decoration: none;
    margin: 3%;
    &:hover{
      background-color: white;
      color: #0F2130;
    }
  }

  .sign-up{
    border: white 2px solid;
    padding: 1%;
    width: 100px;
    color: white;
    text-decoration: none;
    margin: 3%;
    &:hover{
      background-color: white;
      color: #0F2130;
    }
  }

  .links{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-left: 3%;

  }
}
