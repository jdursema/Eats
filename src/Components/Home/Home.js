import React, {Component} from 'react';
import './Home.css';
import { connect } from 'react-redux';
import { fetchRestaurants } from '../../Actions';
import CardConatainer from '../CardContainer/CardContainer';
import { NavLink } from 'react-router-dom';



class Home extends Component {  

  async componentDidMount() {
    //if localstorage.cards then render storage and set it in store else fetch
    try {
      this.props.handleFetchRestaurants();
    } catch (error){
      console.log(error);
    }
  }

  render () {
    return (
      <div className='home'>
        <h1>Welcome to Eats</h1>
        <h3>Find your new favorite restuaurant today!</h3>
        <NavLink to='/login'>Login</NavLink>
      </div>
      
    );
  }
  
}


const mapDispatchToProps = dispatch => ({
  handleFetchRestaurants: () => dispatch(fetchRestaurants())
});

export default connect(null, mapDispatchToProps)(Home);