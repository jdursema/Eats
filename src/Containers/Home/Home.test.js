import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { Home } from './Home';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Home', () => {
  it('should match the snapshot', ()=> {

    const renderedHome = shallow(<Home />);

    expect(renderedHome).toMatchSnapshot();
  });
});