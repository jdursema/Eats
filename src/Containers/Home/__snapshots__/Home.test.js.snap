// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="home-text"
>
        <h1>
                Welcome to Eats
        </h1>
        <h3>
                Look up resaurants and get recommendations based on places you already love. Your next favorite eatery is waiting to be discovered!
        </h3>
        <div
                className="links"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="login"
                        to="/login"
                >
                        Login
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="sign-up"
                        to="/signup"
                >
                        SignUp
                </NavLink>
        </div>
</div>,
      "className": "home",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to Eats
</h1>,
          <h3>
            Look up resaurants and get recommendations based on places you already love. Your next favorite eatery is waiting to be discovered!
</h3>,
          <div
            className="links"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="login"
                        to="/login"
            >
                        Login
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="sign-up"
                        to="/signup"
            >
                        SignUp
            </NavLink>
</div>,
        ],
        "className": "home-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Eats",
          },
          "ref": null,
          "rendered": "Welcome to Eats",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Look up resaurants and get recommendations based on places you already love. Your next favorite eatery is waiting to be discovered!",
          },
          "ref": null,
          "rendered": "Look up resaurants and get recommendations based on places you already love. Your next favorite eatery is waiting to be discovered!",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="login"
                to="/login"
>
                Login
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="sign-up"
                to="/signup"
>
                SignUp
</NavLink>,
            ],
            "className": "links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Login",
                "className": "login",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "SignUp",
                "className": "sign-up",
                "to": "/signup",
              },
              "ref": null,
              "rendered": "SignUp",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="home-text"
>
          <h1>
                    Welcome to Eats
          </h1>
          <h3>
                    Look up resaurants and get recommendations based on places you already love. Your next favorite eatery is waiting to be discovered!
          </h3>
          <div
                    className="links"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="login"
                              to="/login"
                    >
                              Login
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="sign-up"
                              to="/signup"
                    >
                              SignUp
                    </NavLink>
          </div>
</div>,
        "className": "home",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Welcome to Eats
</h1>,
            <h3>
              Look up resaurants and get recommendations based on places you already love. Your next favorite eatery is waiting to be discovered!
</h3>,
            <div
              className="links"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="login"
                            to="/login"
              >
                            Login
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="sign-up"
                            to="/signup"
              >
                            SignUp
              </NavLink>
</div>,
          ],
          "className": "home-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Eats",
            },
            "ref": null,
            "rendered": "Welcome to Eats",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Look up resaurants and get recommendations based on places you already love. Your next favorite eatery is waiting to be discovered!",
            },
            "ref": null,
            "rendered": "Look up resaurants and get recommendations based on places you already love. Your next favorite eatery is waiting to be discovered!",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="login"
                  to="/login"
>
                  Login
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="sign-up"
                  to="/signup"
>
                  SignUp
</NavLink>,
              ],
              "className": "links",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Login",
                  "className": "login",
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "SignUp",
                  "className": "sign-up",
                  "to": "/signup",
                },
                "ref": null,
                "rendered": "SignUp",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
